pkg/kernel/module.go

```diff
@@ -111,12 +111,11 @@ func (mc ModuleConfig) GetType() string {
 // module (see "mon").
 //go:generate mockery --name Module
 type Module interface {
-	// Run the module. If the provided context is canceled you have a
-	// few seconds (configurable with kernel.killTimeout) until your module
-	// is killed (via exit(1)). If you return from Run, it is assumed that
-	// your module is done executing and (depending on the type of your module)
-	// this might trigger a kernel shutdown. If you return an error, a kernel
-	// shutdown is also triggered.
+	// Run the module. If the provided context is canceled you have a  few seconds (configurable with kernel.killTimeout)
+	// until your module is killed (via exit(1)). If you return from Run, it is assumed that your module is done
+	// executing and (depending on the type of your module) this might trigger a kernel shutdown.
+	// If you return an error, all modules in the same stage will receive a closed channel when invoking
+	// `ctx.Done()`. If they ignore it, they will keep running.
```

Comment:
That sounds like a bug to me - if a module fails and another module is stuck as a consequence of that, shouldn't the kernel eventually reach for the kill instead of allowing such an application to hang?